test/plcr.js

```diff
@@ -0,0 +1,603 @@
+
+    // ----------- Create vote --------------
+
+    it('creates vote', async () => {
+      const voteId = await createVote()
+      const vote = await app.getVote.call(voteId)
+      const currentTime = await app.getTimestampExt.call()
+      // checks
+      assert.equal(vote[0].toString(), (currentTime.add(commitDuration)).toString(), "Commit end date should match")
+      assert.equal(vote[1].toString(), (currentTime.add(commitDuration).add(revealDuration)).toString(), "Reveal end date should match")
+      assert.equal(vote[2], false, "Result should match")
+      assert.equal(vote[3], false, "Computed should match")
+      assert.equal(vote[4], "", "Metadata should match")
+      assert.equal(vote[5], 0, "Votes For should match")
+      assert.equal(vote[6], 0, "Votes Against should match")
+      assert.equal(vote[7], 0, "Total votes should match")
+      assert.equal(vote[8], 0, "Slash Pool should match")
+      assert.equal(vote[9], 0, "Paid Reward should match")
+    })
+
+    // ----------- Commit votes --------------
+    it('commits vote For', async () => {
+      const voteOption = true
+      const {voteId, receipt} = await commitVote(voter1, voteOption)
+      // checks
+      const slashPool = minorityBlocSlash.mul(stake).dividedToIntegerBy(pct16(100))
+      assert.isTrue(checkUnlockedPartial(receipt, voter1, app.address, lockId, slashPool))
+      assert.isTrue(checkMovedTokens(receipt, voter1, app.address, slashPool))
+      const userVote = await app.getUserVote.call(voteId, voter1)
+      assert.equal(userVote[0].toString(), lockId, "lockId should match")
+      assert.equal(userVote[1].toString(), stake, "Stake should match")
+      assert.equal(userVote[2], secretHash(true), "Secret Hash should match")
+      assert.equal(userVote[3], false, "Revealed should match")
+      assert.equal(userVote[4], false, "Vote Option should match")
+      assert.equal(userVote[5], false, "Claimed should match")
+      const vote = await app.getVote.call(voteId)
+      assert.equal(vote[5].toString(), 0, "Votes For should match")
+      assert.equal(vote[6].toString(), 0, "Votes against should match")
+      assert.equal(vote[7].toString(), 0, "Total votes should match")
+      assert.equal(vote[8].toString(), minorityBlocSlash.mul(stake).dividedToIntegerBy(pct16(100)), "Slash pool should match")
+      assert.equal(vote[9].toString(), 0, "Paid rewards should match")
+    })
+
+    it('commits vote Against', async () => {
+      const voteOption = false
+      const {voteId, receipt} = await commitVote(voter1, voteOption)
+      // checks
+      const slashPool = minorityBlocSlash.mul(stake).dividedToIntegerBy(pct16(100))
+      assert.isTrue(checkUnlockedPartial(receipt, voter1, app.address, lockId, slashPool))
+      assert.isTrue(checkMovedTokens(receipt, voter1, app.address, slashPool))
+      const userVote = await app.getUserVote.call(voteId, voter1)
+      assert.equal(userVote[0].toString(), lockId, "lockId should match")
```

Comment:
Too similar to the previous test, could probably extract into a function and not repeat so much code?